Core/Core.Platform.Shared/MobileCoreInitializer.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using AeroGear.Mobile.Core.Exception;
+using Xamarin.Forms;
+
+namespace AeroGear.Mobile.Core.Platform
+{
+    public sealed class MobileCoreInitializer
+    {
+        /// <summary>
+        /// Initializes Mobile core for Android.
+        /// </summary>
+        /// <param name="app">Xamarin.Forms.Application object</param>
+        public static void Init(Application app)
+        {
+            Init(app, new Options());
+        }
+
+        /// <summary>
+        /// Initializes Mobile core for Android.
+        /// </summary>
+        /// <param name="app">Xamarin.Forms.Application object</param>
+        /// <param name="options">additional initialization options</param>
+        public static void Init(Application app, Options options)
+        {
+            IPlatformInjector platformInjector = DependencyService.Get<IPlatformInjector>();
+            platformInjector.ExecutingAssembly = app.GetType().Assembly;
+            MobileCore.Init(platformInjector,options);
```

Comment:
so if we add this method to the MobileCore:

```
MobileCore.Init(Assembly, Options)
```

then it means we will be able to initialise the PlatformInjector inside of MobileCore.

From their applications, developers can do:

```
MobileCore.Init(app.GetType().Assembly, options)
```

This means we don't need this class, and then the shared project.

WDYT?

Reply:
Assembly is there only for resources access. Injector can't be initialized in Core project because it is unaware of Platform modules. DependencyService is Xamarin only, Core project is .NET Standard.